### ISSUE TITLE
[Import video from URL] Stopped working after moving server to location that requires proxy settings

### ISSUE THREAD

**Issue by etcy** [WWBN/AVideo#3983]
I had a working installation of AVideo and AVideo-Encoder, on the same Ubuntu machine.
Yesterday, I had to relocat the servers to a different location. Afterwhich, I followed the help from 
https://tutorials.avideo.com/video/18/fix-youphptube-url-after-move-server-location-and-ip

And managed to get the server up and running almost OK.

Now, I have no problems with:
- Viewing videos
- Upload new videos from file through my encoder 

I have problem with:
1. Cannot import new videos from youtube (via encoder site->import video)
 -> [Processing window never close]
 -> log file for Encoder does not record any attempt to start youtube-dl at all
2. Problem adding new youtube links (via avideo site-> embed a video link) 
 -> Seems to take longer than before
 -> Entry is added but Titles and such are not proper (pls see AVideo/avideo.log below for details)
3. Clicking update version leads to very long wait time (I think its might be timing out)
  (but if i let it be, after a while the page loads to say i have the newest version. So this might not be a problem?)

My new location requires accessing the internet thru a proxy server.
I have set them up in the 
1. /etc/environment
2.~/.curlrc
3./etc/wgetrc

I think its related to the proxy settings. Is there somewhere else which I need to set for AVideo to work as before?

I think the biggest issue is import video gets stuck with processing window, and never seems to even proceed to youtube-dl anything. Since the AVideo-Encoder/video/avideo.log does not get updated I have no hint as to where the process went wrong.

**Comment by etcy** [comment]
AVideo/avideo.log

`[10-Nov-2020 14:01:02 UTC] AVideoLog::DEBUG: wget: ERROR the url download but is empty http://XX.XX.XX.XX/AVideo-Encoder/getLinkInfo/aHR0cHM6Ly93d3cueW91dHViZS5jb20vd2F0Y2g/dj1uTTlzT2ZGS25GQQ==, /var/www/html/AVideo/videos/cache/YPTurl_get_contents/3737f3a02e63291c895f8d3e4e89b139 SCRIPT_NAME: /AVideo/objects/videoAddNew.json.php
[10-Nov-2020 14:02:02 UTC] PHP Notice:  Trying to get property 'title' of non-object in /var/www/html/AVideo/objects/videoAddNew.json.php on line 48
[10-Nov-2020 14:02:02 UTC] PHP Notice:  Trying to get property 'title' of non-object in /var/www/html/AVideo/objects/videoAddNew.json.php on line 49
[10-Nov-2020 14:02:02 UTC] PHP Notice:  Trying to get property 'duration' of non-object in /var/www/html/AVideo/objects/videoAddNew.json.php on line 50
[10-Nov-2020 14:02:02 UTC] PHP Notice:  Trying to get property 'description' of non-object in /var/www/html/AVideo/objects/videoAddNew.json.php on line 51
[10-Nov-2020 14:02:02 UTC] PHP Notice:  Trying to get property 'thumbs64' of non-object in /var/www/html/AVideo/objects/videoAddNew.json.php on line 52
[10-Nov-2020 14:02:02 UTC] AVideoLog::DEBUG: Warning: Slow process detected [/var/www/html/AVideo/objects/videoAddNew.json.php] On  Line 85 takes 120.1984 seconds to complete. /var/www/html/AVideo/objects/videoAddNew.json.php SCRIPT_NAME: /AVideo/objects/videoAddNew.json.php
[10-Nov-2020 14:02:02 UTC] AVideoLog::DEBUG: Video::save (5faa9d5a645f1) Saved id = 1  SCRIPT_NAME: /AVideo/objects/videoAddNew.json.php
[10-Nov-2020 14:24:32 UTC] AVideoLog::DEBUG: Start Login Request SCRIPT_NAME: /AVideo/objects/login.json.php
[10-Nov-2020 14:24:32 UTC] AVideoLog::DEBUG: redirectUri: http://XX.XX.XX.XX/AVideo/ SCRIPT_NAME: /AVideo/objects/login.json.php
[10-Nov-2020 14:24:32 UTC] AVideoLog::DEBUG: same redirectUri: http://XX.XX.XX.XX/AVideo/ SCRIPT_NAME: /AVideo/objects/login.json.php
`

I tried to use wget and youtube-dl directly from command line and they work as expected but somehow, the wget within Avideo dun seem to work with the system proxy settings?

**Comment by etcy** [comment]
About (encoder site->import video) issue, after waiting for a looong time (timeout) finally something is showing in the log file.
Looking into the encoder log file: (AVideo-Encoder/videos/avideo.log)

Currently, the last entry shows:
Verification Start http://X.X.X.X/AVideo/
Verification Creating the Cache http://X.X.X.X/AVideo/
Verification Response (https://search.avideo.com/verify.php?url=http%3A%2F%2FX.X.X.X%2FAVideo%2F): 
getTitleFromLink: Get Title Error: LC_ALL=en_US.UTF-8 /usr/local/bin/youtube-dl  --no-check-certificate --no-playlist --force-ipv4 --skip-download -e "https://www.youtube.com/watch?v=nM9sOfFKnFA" 
Array
(
    [0] => ERROR: Unable to download webpage: <urlopen error [Errno 110] Connection timed out> (caused by URLError(TimeoutError(110, 'Connection timed out'),))
)

[10-Nov-2020 20:01:42 UTC] youtubeDl::addVideo We could not get the title () of your video (https://www.youtube.com/watch?v=nM9sOfFKnFA)
[10-Nov-2020 20:27:11 UTC] getTitleFromLink: Get Title Error: LC_ALL=en_US.UTF-8 /usr/local/bin/youtube-dl  --no-check-certificate --no-playlist --force-ipv4 --skip-download -e "https://www.youtube.com/watch?v=nM9sOfFKnFA" 
Array
(
    [0] => ERROR: Unable to download webpage: <urlopen error [Errno 110] Connection timed out> (caused by URLError(TimeoutError(110, 'Connection timed out'),))
)

[10-Nov-2020 20:28:11 UTC] getTitleFromLink: Get Title Error: LC_ALL=en_US.UTF-8 /usr/local/bin/youtube-dl  --no-check-certificate --no-playlist --force-ipv4 --skip-download -e "https://www.youtube.com/watch?v=nM9sOfFKnFA" 
Array
(
    [0] => ERROR: Unable to download webpage: <urlopen error [Errno 110] Connection timed out> (caused by URLError(TimeoutError(110, 'Connection timed out'),))
)

----------------
So from this log, I have two questions.

1. Compare the older entries when it was working, I noticed the 3 line of Verification Response, was longer with MSG details. 
For example:
Verification Response (https://search.avideo.com/verify.php?url=http%3A%2F%2FX.X.X.X%2FAVideo%2F): {"msg":["Start","http:\/\/X.X.X.X\/AVideo\/ is a private IP"],"verified":true,"status":false}

Is this a problem?

2. Why is youtube-dl timing out. I executed youtube-dl on the commandline, with the same url and it works.

**Comment by etcy** [comment]
After rethinking my problem and looking at the logs again, I figure that my issue with getting stuck at "processing" for a long is 

1. Unlikely to be due to Verification Response

2. More likely to be due to youtube-dl timing out. And since my setup was working before relocation, the only things that changed were server IP (dealt with) and a need for proxy setting to access the internet.

So I figured, why not hardcode the proxy setting in the lines that execute youtub-dl. So I did a search on "--no-check-certificate" which lead me to the following file:

/www/html/AVideo-Encoder/objects/Encoder.php

There were several lines that ran the cmd, so I added "--proxy http://X.X.X.X:8080/" like the following and the I could once again import video from youtube.

$cmd = "LC_ALL=en_US.UTF-8 " . self::getYouTubeDLCommand() . "  --proxy http://X.X.X.X:8080/ --no-check-certificate --no-playlist --force-ipv4 --skip-download -e \"{$link}\"";

Its a very dirty approach and I wonder what else is broken for me if apache/php is not seeing the proxy settings I made in /etc/enviroment.

If anyone can share some wisdom with me....
